src/xml_db/dump/mod.rs

```diff
@@ -203,8 +203,13 @@ impl DumpXml for CustomData {
     ) -> Result<(), xml::writer::Error> {
         writer.write(WriterEvent::start_element("CustomData"))?;
 
-        for item in &self.items {
+        for (key, item) in &self.items {
+            writer.write(WriterEvent::start_element("Item"))?;
+
+            SimpleTag("Key", key).dump_xml(writer, inner_cipher)?;
```

Comment:
@varjolintu how about we leave the key in the `CustomDataItem`, even if we index the items in the `CustomData` struct? I feel like it would keep us closer to the data format, and would still provide for easier access of the items. It would also avoid the need to update the dumping code.

Reply:
It would keep us closer to the database format, but I think it would decrease the usability. More comments below in the other reply.